vault/resource_generic_secret.go

```diff
@@ -104,14 +118,50 @@ func genericSecretResourceWrite(d *schema.ResourceData, meta interface{}) error
 	}
 
 	log.Printf("[DEBUG] Writing generic Vault secret to %s", path)
-	_, err = client.Logical().Write(path, data)
+	secret, err := client.Logical().Write(path, data)
 	if err != nil {
 		return fmt.Errorf("error writing to Vault: %s", err)
 	}
 
 	d.SetId(path)
 
-	return genericSecretResourceRead(d, meta)
+	captureResponse := d.Get("capture_response").(bool)
+	if captureResponse {
+		log.Printf("[DEBUG] Capture response is set.")
+		if secret == nil {
+			log.Printf("[WARN] write to (%s) returned empty response", path)
+			return nil
+		}
+		jsonDataBytes, err := json.Marshal(secret.Data)
+		log.Printf("[DEBUG] Secret is %s", jsonDataBytes)
```

Comment:
I don't know if you want to log this even under `DEBUG`.

Reply:
Oops, meant to remove that.  :-O  Will take care of that.